src/rpy_diffusive_flux.cc

```diff
@@ -83,13 +103,14 @@ void RPYDiffusiveFlux::compute(std::shared_ptr<GrandPotential> grand, std::share
  
                 //RPY flux calculation   
                 const double d_ij = a_i + a_j;
-                
+                const double x_int = std::min<double>(x, max_x);
+            
                 //To remove the concern about the lower bound value spill over the buffer sites
-                const int ig_low = std::ceil((std::abs(x-d_ij)-mesh->lower_bound())/mesh->step());
-                const int ig_high = mesh->bin(x+d_ij);
+                const double cutoff = std::get<1>(g.getBounds())*d_ij;
```

Comment:
You've already hard coded the diameter, so I don't think you should multiply by it here.
```suggestion
                const double cutoff = std::get<1>(g.getBounds());
```